### PR TITLE
[EPROD-849] Direct ICRC token transfer instead of spender canister

### PR DESCRIPTION
**Pull request by F3kilo**
## Issue ticket

Issue ticket link:
https://infinityswap.atlassian.net/browse/EPROD-849

### Code conventions

- [X] I have performed a self-review of my code
- [X] Every new function is documented
- [X] Object names are auto explicative
- [X] The PR follows the [project conventions](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/23330839/Conventions)
- [X] The code follows the [style guidelines of the project](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/24444929/Rust+Conventions)
- [X] The code follows the [project security best practices](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/35225620/Security+Considerations)

### Security 

- [X] The PR does not break APIs backward compatibility
- [X] The PR does not break the stable storage backward compatibility

### Testing 

- [X] Every function is properly unit tested
- [X] I have added integration tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,41 +21,41 @@ async-trait = "0.1"
...
-ethereum-json-rpc-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "ethereum-json-rpc-client", tag = "v0.17.x" }
+eth-signer = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "eth-signer", branch = "icrc2_minter_without_spender" }
+ethereum-json-rpc-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "ethereum-json-rpc-client", branch = "icrc2_minter_without_spender" }
 ethereum-types = "0.14"
 ethers-core = "2.0"
-evm-canister-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "evm-canister-client", tag = "v0.17.x" }
-icrc-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "icrc-client", tag = "v0.17.x" }
+evm-canister-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "evm-canister-client", branch = "icrc2_minter_without_spender" }
+icrc-client = { git = "https://github.com/bitfinity-network/bitfinity-evm-sdk", package = "icrc-client", branch = "icrc2_minter_without_spender" }
```
Comment (F3kilo) [inline Cargo.toml:31]: Replace with tag

**Comment by F3kilo** [comment]
Add integration test with broken ICRC token.

File: src/icrc2-minter/src/state.rs [hunk 2]
```diff
@@ -87,9 +86,8 @@ impl Default for Settings {
             owner: Principal::anonymous(),
             evm_principal: Principal::anonymous(),
             signing_strategy: SigningStrategy::Local {
-                private_key: [1u8; 32],
+                private_key: [218u8; 32],
```
Comment (blutooth) [inline src/icrc2-minter/src/state.rs:89]: What is this for?

**Review by blutooth — APPROVED** [review]

File: src/icrc2-minter/src/state.rs (hunk 2 above)
Comment (F3kilo) [reply src/icrc2-minter/src/state.rs:89]: To prevent accidental coincidence of private keys of different test cansiters.

**Pull request merged**